### ISSUE TITLE
Deleting fields and values after deletion not happening in background

### ISSUE THREAD

**Issue by dsilvam** [dotCMS/core#16939]
Some versions ago we did some work to [make the cleaning of a field values when deleting the field to run in a background job ](https://github.com/dotCMS/core/issues/8793). 

When creating the new ContentType and Field API we stopped using the background job for field deletion and this is now happening on a synched fashion, hence the [referred issue](https://github.com/dotCMS/core/issues/8793) would occur again. 

In the mentioned [background job](https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/quartz/job/DeleteFieldJob.java#L60) we used to include the entire field deletion process - one field at a time - but we now suggest to only make the cleaning of the field values async, because changing to start using the Job again implies making some changes; at REST API level, since the answer would not be synced anymore; and at background job level, to adjust it to process more than one field at a time. 

### Update 2019-10-14
The field deletion should happen in two stages if possible.  The first stage is the deletion of the field from the database - this can happen instantaneously, which should also kick off a structure reindex.

The second stage of deletion is the cleaning of the DB and assets and this should happen as a job.  Because of the asynchronous nature of this, when we start the job to clean the DB and assets,we should include the timestamp of the field delete action in the job map and only clean/update those contentlets whose timestamps are less than the timestamp in the clean/update job.  Additionally, the Field clean/update job needs to be StatefulJobs so that only one can be running at a time.

**Comment by dsilvam** [comment]
Related to: https://github.com/dotCMS/core/issues/13718

**Comment by wezell** [comment]
Related to #17448

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/core/pull/17522/files

**Comment by dsilvam** [comment]
Passed Internal QA Release: returning immediately after deleting the field in the UI and background job taking care of the cleaning of the field values. Tested with "Document" content type from Auth dataset. Also verified that db column assigned to field was properly cleaned.

**Comment by dsilvam** [comment]
Important note: got the warning about Transaction broken, but I think this is expected since we don't have the code changes that deal with it. 

````
15:35:32.491  WARN  db.LocalTransaction - Transaction broken - Connection that started the transaction is not the same as the one who is commiting
15:35:32.491  WARN  db.LocalTransaction -     java.lang.Thread.getStackTrace(Thread.java:1559)
15:35:32.491  WARN  db.LocalTransaction -     com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:57)
15:35:32.491  WARN  db.LocalTransaction -     com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23)
15:35:32.491  WARN  db.LocalTransaction -     com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41)
15:35:32.491  WARN  db.LocalTransaction -     com.dotcms.contenttype.business.ContentTypeFieldLayoutAPIImpl.deleteField(ContentTypeFieldLayoutAPIImpl.java:146)
15:35:32.491  WARN  db.LocalTransaction -     com.dotcms.rest.api.v3.contenttype.FieldResource.deleteFields(FieldResource.java:211)
1
```

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.1 // Postgres // FF

**Comment by nollymar** [comment]
New PR https://github.com/dotCMS/core/pull/17598 to fix integration test in Oracle

**Comment by dsilvam** [comment]
Broken by this change: https://github.com/dotCMS/core/issues/17570

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/17609

**Comment by wezell** [comment]
Does the code in this issue #17591 fix the broken transaction?

**Comment by dsilvam** [comment]
**Note to QA**: To test this please delete the "Headline" field of the "News" content type and run this sql:
`select text4
from contentlet c join structure s on c.structure_inode = s.inode
where s.name = 'News'
` and make sure it returns empty values

**Comment by dsilvam** [comment]
**Passed Internal QA**:Clearing deleted field value as expected. Checked by using sql against the column in the contentlet table mapped to the deleted field.

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.2 // Postgres... Following the comment for QA, after deleting the field and run the query we are returning 25 empty rows.
